src/Ethereum/ABI/ParamFactory.cpp

```diff
@@ -74,9 +112,31 @@ std::shared_ptr<ParamBase> ParamFactory::make(const std::string& type) {
     return param;
 }
 
+std::string joinArrayElems(const std::vector<std::string>& strings, bool quoteValues) {
```

Comment:
just return json::array::dump()?

Reply:
It's more elegant with json.  I had to tweak it a bit to avoid quotes on simple values.